tests/data/get_posts.php

```diff
@@ -7,6 +7,42 @@
 use function PHPStan\Testing\assertType;
 
 assertType('array<int, WP_Post>', get_posts());
-assertType('array<int, WP_Post>', get_posts(['fields' => 'all']));
+assertType('array<int, WP_Post>', get_posts(['key' => 'value']));
+assertType('array<int, WP_Post>', get_posts(['fields' => '']));
 assertType('array<int, int>', get_posts(['fields' => 'ids']));
 assertType('array<int, int>', get_posts(['fields' => 'id=>parent']));
+assertType('array<int, WP_Post>', get_posts(['fields' => 'Hello']));
+
+// Non constant array
```

Comment:
```suggestion
// Nonconstant array
```

Please see https://www.merriam-webster.com/dictionary/nonconstant